### PR TITLE
feat(bun): add support for bun yarn.lock files

### LINKED ISSUE #15: Support parsing Bun lockfiles
@robertohuertasm 

Hey Robert. 

By default Bun creates a binary lockfile `bun.lockb` but can represent this as a Yarn v1 lockfile with `bun bun.lockb` or `bun install --yarn`. More info: https://bun.sh/docs/install/lockfile

However, when parsing Bun's yarn compatible output, this crate fails to parse it, _and I don't know why_. For example, this is from Bun:

```
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1
# bun ./bun.lockb --hash: 5572C9522DE896FD-4a82b28bb592ae3b-4B24ADDA3457FA29-1c299c64c91cddb0


typescript@4.8.4:
  version "4.8.4"
  resolved "https://registry.npmjs.org/typescript/-/typescript-4.8.4.tgz"
  integrity sha512-QCh+85mCy+h0IGff8r5XWzOVSbBO+KfeYrMQh7NJ58QujwcE22u+NUSmUxqF+un70P9GXKxa2HCNiTTMJknyjQ==
```

But it errors with:

```
VerboseError { errors: [("descriptor does not end with : followed by newline", Nom(Fail)), ....
```

The format of this file looks correct, so I'm guessing this is barfing on the `# bun ...` comment?

### PR DESCRIPTION
**Pull request by milesj**
@robertohuertasm Couldn't figure out how to get the `is_version` to work, but all tests pass.

Fix #15

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -316,28 +320,29 @@ fn entry_version(input: &str) -> Res<&str, EntryItem> {
...
+// #[allow(clippy::needless_pass_by_value)]
+// fn is_version<T, E: nom::error::ParseError<T>>(input: T) -> IResult<T, T, E>
+// where
+//     T: nom::InputTakeAtPosition,
+//     <T as nom::InputTakeAtPosition>::Item: AsChar,
+// {
+//     input.split_at_position1_complete(
+//         |item| {
+//             let c: char = item.as_char();
+//             !(c == '.' || c == '-' || c.is_alphanum())
```
Comment (robertohuertasm) [inline src/lib.rs:341]: Just by adding the new non alphanumeric chars would do the trick: ` !(c == '.' || c == '-' || c == '@' || c == '/' || c == ':' || c.is_alphanum())`

**Review by robertohuertasm — COMMENTED** [review]
Thanks for this PR! I think we can live with these 3 lines for the header. Even though the yarn.lock file example in the Yarn website has only 2 and no spaces (see [here](https://classic.yarnpkg.com/lang/en/docs/yarn-lock/)), that doesn't seem to what yarn ends generating and it always adds at least one space.

Regarding the version, see my comment. In any case, I think your implementation is good. My only concern is that, so far, version was supposed to be returning always `x.x.x` or similar. I was relying on that mechanism in another project I have that depends on this (that's why I initially created this crate). Let me review how this would affect that project and I'll come back to you.

I'm inclined to keep the current solution you provided, but leave me a few days more to consider before merging this.

Again, thanks a lot for your contribution ❤️

**Comment by milesj** [comment]
@robertohuertasm As an alternative. We can probably wrap this in a `bun` feature or function argument? That way version parsing stays strict by default.

**Comment by robertohuertasm** [comment]
@milesj that's also a good idea. Anyways, I've checked if the change could break anything on my side and it's safe. I'd probably defer using a feature until we find any issues with the current approach.

File: src/lib.rs (hunk 1 above)
Comment (robertohuertasm) [reply src/lib.rs:341]: @milesj would that suggestion be ok? or are you foreseeing more needed chars? If that's the case, I'd go with your current implementation and completely delete the `is_version` commented function.

**Review by robertohuertasm — APPROVED** [review]
Let's merge this. Thanks for the change!

**Comment by milesj** [comment]
Sorry was busy this weekend, but thanks for merging!

**Pull request merged**